lib/puppet_x/puppetlabs/strings/yard/handlers/host_class_handler.rb

```diff
@@ -8,6 +8,21 @@ class PuppetX::PuppetLabs::Strings::YARD::Handlers::HostClassHandler < PuppetX::
         param_tuple << ( a[1].nil? ? nil : a[1].source )
       end
     end
+    tp = Puppet::Pops::Types::TypeParser.new
+    param_type_info = []
+    # Ensure that this pop object has parameters
+    if statement.pops_obj.respond_to? :parameters
```

Comment:
hm, not the best way to ensure this only deals with parameters of things we want to document. Much better to test the type of the pops_obj. Some ruby-ists think "responds_to?" is a great idea and use it judiciously, but it should really be reserved for last resort / meta programming / adapting to versions of an imlementation etc. as it leads to horrible spaghetti and opens for surprising behavior.